src/config.rs

```diff
@@ -1330,11 +1331,18 @@ impl TiKvConfig {
         }
 
         if last_cfg.storage.data_dir != self.storage.data_dir {
-            return Err(format!(
-                "storage data dir have been changed, former data dir is {}, \
-                 current data dir is {}, please check if it is expected.",
-                last_cfg.storage.data_dir, self.storage.data_dir
-            ));
+            // In tikv 3.0 the default value of storage.data-dir changed
+            // from "" to "./"
+            let using_default_after_upgrade =
```

Comment:
can we add a test that using empty before?

Reply:
I may not understand your question, but on line 1337, `last_cfg.storage.data_dir.is_empty()` is testing that their previous configuration used the empty string.